test/cg.jl

```diff
@@ -0,0 +1,18 @@
+using DFTK
+using Test
+using IterativeSolvers
+using LinearAlgebra: norm
+
+@testset "Test conjugate gradient method" begin
+    n = 10
+    A = rand(n, n)
+    A = A' * A + I
+    b = rand(n)
+    tol = 1e-10
+
+    x, ch = DFTK.CG(A, b; tol, maxiter=2n)
+
+    @test norm(A*x - b) ≤ 1e-10
+    @test ch.isconverged
+    @test ch.niter == n+1
+end
```

Comment:
Also test type stability.

Reply:
That's not a type stability test as Float64 will be the default type. See also https://docs.julialang.org/en/v1/stdlib/Test/#Test.@inferred

Reply:
I would use Float32 for a small test and see if you get that back.

Reply:
well then go for lower. It's just to ensure type stability (you will need two tests though!)

Reply:
Oh sorry. You can do two things:
- Have a test in Float64 to full accuracy (done :smile:)
- Have a test in Float32, check you get Float32 back. Use `@inferred` on the CG to check the compiler manages to infer the Float32. If this does not work, scratch the `@inferred`.

Reply:
Yes, but the CG actually returns several things, among which the number of iterations which have no reason to be type dependent, so I should create an intermediate function that returns for instance only the residual or something like that right ?

Reply:
Yes, the residual or (what I initially had in mind) the solution.